server/src/main/java/org/elasticsearch/script/field/vectors/ByteMultiDenseVectorDocValuesField.java

```diff
@@ -138,5 +138,15 @@ public byte[] next() {
             idx++;
             return buffer;
         }
+
+        @Override
+        public Iterator<byte[]> copy() {
+            return new ByteVectorIterator(vectorValues, buffer, size);
```

Comment:
Is it ok to reuse the buffer instead of creating a new one for each copy?

Reply:
@jimczi my concern is when users have an iterator they are iterating. But WHILE iterating, they conduct a `maxSimDotProduct`. 

@jdconrad could you comment here? Basically, I don't know if the same class instance for a field is usable in a script and thus my concern here is warranted.

Reply:
In short, yes @jimczi reusing the buffer is a bad idea. Fixed that and now copy will create a new buffer. This way all iterators have their own buffer, but for typical maxsim scoring, we can reuse the buffer there.

Reply:
Sorry for the delay in commenting. I agree that we should not re-use the buffer given that while it's probably safe for expert users, I worry about future code changes that affect the re-usability.